### PR TITLE
Fix: Quoted names breaking scripts

### LINKED ISSUE #17: String variables including quotes need escape for osascript
Tracks, playlists, and albums are being ignored by `osascript` if they have a single or double quote in their names hance not starting to play/not being favorited/etc. These probably need some type of extra character escape magic, or something else.

E.g. these won't start playing at all:

Screenshot 2024-07-25 at 18 16 13

### PR DESCRIPTION
**Pull request by p5quared**
There is an issue where names with quotes in them are not able to be used.

I fixed this by escaping them:
1. `'` -> `'\\''`
2. `"` -> `\\"`

Closes #17

### CODE REVIEW

**Comment by p5quared** [comment]
@ducktordanny Hey would you mind taking a look at this? Just testing if you can find a song/playlist/etc that doesn't play?

File: lua/apple-music/init.lua [hunk 1]
```diff
@@ -44,6 +44,11 @@ local execute = function(cmd)
 	return pcall(exe, cmd)
 end
 
+--- Escape single and double quotes from names
+local sanitize_name = function(name)
+	return name:gsub("'", "'\\''"):gsub('"', '\\"')
```
Comment (ducktordanny) [inline lua/apple-music/init.lua:49]: I still found an edge case that this change does not handle. When the song has an escape character (`\`) before a single or double quote it still breaks it. I've found the following song for this:

Screenshot 2024-07-27 at 08 38 26

Or we can just create a playlist that has these types of magic characters (not that I'd imagine anyone doing this):

Screenshot 2024-07-27 at 08 39 54

For this, to be solved I'd suggest adding an extra gsub before the other ones, e.g.:

```
diff --git a/lua/apple-music/init.lua b/lua/apple-music/init.lua
index e8e4bf4..6dca0e3 100644
--- a/lua/apple-music/init.lua
+++ b/lua/apple-music/init.lua
@@ -46,7 +46,7 @@ end
 
 --- Escape single and double quotes from names
 local sanitize_name = function(name)
-	return name:gsub("'", "'\\''"):gsub('"', '\\"')
+	return name:gsub("\\", "\\\\"):gsub("'", "'\\''"):gsub('"', '\\"')
 end
 
 local grab_major_os_version = function()
```

**Review by ducktordanny — CHANGES_REQUESTED** [review]

File: lua/apple-music/init.lua (hunk 1 above)
Comment (p5quared) [reply lua/apple-music/init.lua:49]: Haha 😆 !! From the album "Escape Sequence" really!? This is actually really funny to me. It's tempting to punish the artist for choosing such a cruel name.

Thanks for finding this. 👍 

An alternate approach I found was using `quoted form of` from applescript but I didn't really wanna mess around with it for too long. I'll apply this and test it later today and put it out tonight. ✅

Comment (ducktordanny) [reply lua/apple-music/init.lua:49]: Yeah, I think it's a good enough workaround, I was also thinking of something like this, not sure if there are better ways to do this though. LGTM 🫡

**Pull request merged**